src/scraping/collector.py

```
@@ -0,0 +1,123 @@
+    MAX_RETRIES
+)
+
+class DataCollector:
+    def __init__(self):
+        self.auth = aiohttp.BasicAuth(BRIGHTDATA_USERNAME, BRIGHTDATA_PASSWORD)
+        self.base_url = f"https://{BRIGHTDATA_ZONE}.brightdata.com"
+        self.session = None
+        self.batch_size = SCRAPING_BATCH_SIZE
+        self.delay = SCRAPING_DELAY
+
+    async def __aenter__(self):
```

Ruff (0.8.2)

Missing return type annotation for special method `__aenter__`

(ANN204)